api-list.go

```diff
@@ -629,6 +632,15 @@ type ListObjectsOptions struct {
 
 	// Use the deprecated list objects V1 API
 	UseV1 bool
+
+	headers http.Header
+}
+
+// Set adds a key value pair to the options. The
+// key-value pair will be part of the HTTP GET request
+// headers.
+func (o *ListObjectsOptions) Set(key, value string) {
+	o.headers.Set(key, value)
```

Comment:
This will crash if o.headers is not initialized. You nedd to initialize it here I guess.